streaming-binlog/src/main/java/org/openmrs/analytics/FhirUriGenerator.java

```diff
@@ -0,0 +1,95 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package org.openmrs.analytics;
+import java.util.Map;
+
+
+import org.apache.camel.Exchange;
+import org.apache.camel.Processor;
+import org.apache.camel.component.debezium.DebeziumConstants;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * A Debezium events to FHIR URI mapper
+ * can be configured to use the utils/fhir2_atom_feed_config.json mapper
+ */
+public class FhirUriGenerator implements Processor {
+
+    private static final Logger LOG = LoggerFactory.getLogger(DebeziumListener.class);
+    private static final String EXPECTED_BODY_FORMAT = "{\"fhirResourceUri\":%s}";
+
+    public void process(Exchange exchange) throws Exception {
+        final Map payload = exchange.getMessage().getBody(Map.class);
+        final String headerIdentifier = exchange.getMessage().getHeader(DebeziumConstants.HEADER_IDENTIFIER, String.class);
+        final String table = headerIdentifier.substring(headerIdentifier.lastIndexOf('.') + 1).trim();
+
+        LOG.info("Processing Table --> " +table );
+        // TODO make this configurable using utils/fhir2_atom_feed_config.json
+        switch (table) {
+            case "obs":
+                exchange.getIn().setHeader("fhirResourceUri", "/Observation/"+payload.get("uuid").toString());
```

Comment:
Do we always get `uuid` in the `payload`?

Reply:
Yes! All OpenMRS tables have enforced this!

Reply:
Right; I was mostly asking about what Debezium payload has, i.e., whether it has `uuid` in an update message if `uuid` is not changed. I think this is a minor issue and I can check Debezium payloads myself once we merge your PR; marking as resolved.